modules/llamacpp_model.py

```diff
@@ -11,6 +11,32 @@
 from modules.text_generation import get_max_prompt_length
 
 
```

Comment:
Are you sure all of these quant types work with llama.cpp?

I did some search here: [#6168 ](https://github.com/oobabooga/text-generation-webui/issues/6168#issuecomment-2198087925)
From what I saw in llama.cpp list of supported cache types is shorter than list of all quant types (there were no K-quants in supported cache quantizations).

I checked llama.cpp code a bit more now and found this: [common.cpp](https://github.com/ggerganov/llama.cpp/blob/26a8406ba9198eb6fdd8329fa717555b4f77f05f/common/common.cpp#L1018)
From what I understand this function kv_cache_type_from_str determines type for KV cache for llama.cpp.
And it seems to allow less types, it allows only: "f32", "f16", "bf16", "q8_0", "q4_0", "q4_0", "q4_1", "iq4_nl", "q5_0", "q5_1".
In other cases it fails with "Unsupported cache type".

Did you check if all the cache types for llama.cpp added in PR work? Especially those not on the list in llama.cpp code (like q6_k or q4_k)?

Reply:
Right, thanks for pointing that out, part of finalizing this PR is checking to ensure the KV cache matrix is correct. Additionally, many of these are only supported if you build with `GGML_CUDA_FA_ALL_QUANTS`, which I'm not sure if we have enabled on our wheel. I'll ensure this is all to code before I mark ready.